src/main/com/diana/restaurant/queue/QueueManager.java

```diff
@@ -0,0 +1,58 @@
+package com.diana.restaurant.queue;
+
+import com.diana.restaurant.entities.EntityPriorityForQueue;
+import com.diana.restaurant.entities.Order;
+import com.diana.restaurant.enums.Queues;
+import com.diana.restaurant.exceptions.KeyNotFoundException;
+import com.diana.restaurant.exceptions.services.NullEntityException;
+import com.diana.restaurant.util.QueueManagerUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+public class QueueManager {
+  private static QueueManager instance;
+  private final Map<Queues, Queue> instanceMap = new HashMap<>();
```

Comment:
Maybe a `ConcurrentMap` is better option?

Reply:
Yes, I investigated and using HashMaps is not a very good option when working with threads. ConcurrentHashMap is thread safe, and doesn't create synchronization issues. As well as only locking a portion of the map instead of the entire map when using.